### PR TITLE
Pouch / Cloud Planning integration (WIP)

### PR DESCRIPTION
**Pull request by lindner**
- switch from protocol override (which led to weird Storage Keys like:
      pouchdb://arcs-storage.firebaseio.com/...
- stop using private methods in Planificator.  Instantiate a separate
      storageprovider base.
- revamp storagekey construction.

### CODE REVIEW

**Comment by lindner** [comment]
This is reviewable, but includes other PRs that can be split off.

File: runtime/ts/plan/planificator.ts [hunk 1]
```diff
@@ -115,39 +115,37 @@ export class Planificator {
...
 
-  private static async _initStore(arc, {userid, protocol, arcKey}): Promise<StorageProviderBase> {
+  private static async _initStore(arc: Arc, {userid, storageKeyBase, arcKey}): Promise<StorageProviderBase> {
     assert(userid, 'Missing user id.');
-    let storage = arc.storageProviderFactory._storageForKey(arc.storageKey);
-    const storageKey = storage.parseStringAsKey(arc.storageKey);
-    if (protocol) {
-      storageKey.protocol = protocol;
+    assert(storageKeyBase, 'Missing storageKeyBase');
+    if (!arcKey) {
```
Comment (mmandlis) [inline runtime/ts/plan/planificator.ts]: i think `arcKey` is never really passed - see `arcKey: null` above (sorry, it is probably a remainder of my refactoring), but `arc.id` is not the right key to use, we'd still need to extract the key from arc.storageKey's location using a regex.

Comment (lindner) [reply runtime/ts/plan/planificator.ts]: Okay, but arc.storageKey is not public, and we shouldn't be relying on the underlying storage.

Should we add a method on within `Arc` that computes this for us?  Something else?

Comment (mmandlis) [reply runtime/ts/plan/planificator.ts]: I'm not sure this should be `Arc` that does it, but maybe the `StorageKey` class? @shans WDYT?

**Comment by lindner** [comment]
outdated.

**Pull request closed without merging**